merge/update.go

```diff
@@ -200,13 +199,10 @@ func (s *Updater) Apply(liveObject, configObject *typed.TypedValue, version fiel
 	if err != nil {
 		return nil, fieldpath.ManagedFields{}, fmt.Errorf("failed to prune fields: %v", err)
 	}
-	managers, compare, err := s.update(liveObject, newObject, version, managers, manager, force)
+	managers, _, err = s.update(liveObject, newObject, version, managers, manager, force)
 	if err != nil {
 		return nil, fieldpath.ManagedFields{}, err
 	}
-	if compare.IsSame() {
-		newObject = nil
-	}
```

Comment:
I think this is the nil you are talking about?

The ManagedFieldsUpdater is the last in the FieldManager chain. It seems responsible for handling the returned nil:

https://github.com/kubernetes/kubernetes/blob/2057a48ee5f6eab586e731f8544c793b03211cf1/staging/src/k8s.io/apimachinery/pkg/util/managedfields/internal/managedfieldsupdater.go#L70-L82

Reply:
So it doesnt seem like at least for k8s usage this is that risky. The nil is bubbled through the field manager chain until it is actualized into the old value.

Reply:
So object is always going to be non-nil, so the timestamp will always be updated, should we compare the value and see if they changed?

Reply:
Instead of doing a compare here, I could do a `value.Equals` and keep the exact same behavior, which would be a little less risky and easier to backport.

Reply:
I dont think it matters to return nil here or not. The nil only goes through StipMetaManager and ManagedFieldsUpdater before becoming non-nil, and neither of those care about it being the same object.

Reply:
I see the timestamp update is conditional on the object returned being nil, but I think that can be made unconditional due to the no-op modulo timestamp check inside the store.